### PR TITLE
chore: parallel Maven reactor build (-T1C) + wrapper bump 3.9.16

### PR DESCRIPTION
**Pull request by cuioss-oliver**
## Summary

Speeds up CI and local builds by enabling a **parallel Maven reactor** (`-T1C`).

The CI Maven Build workflow runs three sequential full builds — matrix `build` (JDK 21+25) → `sonar-build` (a second full `verify`) → `deploy-snapshot` (a third full build) — and **every one is single-threaded**. The job DAG lives in the external org reusable workflow and isn't changeable from this repo, but all three jobs invoke `./mvnw`, which reads `.mvn/maven.config` (previously empty). Populating it with `-T1C` parallelises all of them, plus the integration/e2e `-am` upstream builds.

Local `clean verify` drops from **~2:14 to ~1:00** (parallel `-Pcoverage` build measured at 1:08).

## Changes

- **`.mvn/maven.config`**: add `-T1C` (one thread per core).
- **`.mvn/wrapper/maven-wrapper.properties`**: bump Maven `3.9.11` → `3.9.16` (latest 3.9.x; enforcer requires ≥ 3.9.6). Maven 4.x is still RC and not adopted.

### Test hardening for stable parallel runs
- **rest-processors HTTP tests** pinned to **HTTP/1.1** (`RestApiGatewayProcessor`, `Management`, `Attachments`, `RequestUtils`, `Status`, `JettyServerManager`). Under concurrent load the JDK HttpClient's HTTP/2 cleartext negotiation against embedded Jetty raced (`received no bytes`/`EOFException`); HTTP/1.1 is deterministic.
- **`GatewayRequestHandlerTest`**: retry-on-transient-`IOException` with a fresh client + backoff; treat a connection-close on malformed URIs (e.g. `%00`, which Jetty correctly rejects by closing) as a valid rejection rather than a flake.

### Coverage top-ups (keep local `-Pcoverage` 0.80 gate green)
- **`HttpRequestContainerTest`**: record `equals`/`hashCode` one-field-differs contract.
- **`GatewayProxyServletTest`**: `buildMergedRoutesArray` + `isLocalhostTarget` edge cases.

## Verification
- Full `-Pcoverage` (all modules, BUNDLE 0.80 gate) **green** on the exact committed set — 1:08 min.
- Multiple clean `verify` runs green (~1 min vs 2:14 baseline).
- Flaky HTTP test stabilised (10/10 under load).

## Isolation notes
Surefire/Failsafe keep defaults (`forkCount=1`, `reuseForks=true`) — each module's tests already run in their own forked JVM, so `-T` parallelises across modules without in-JVM test concurrency. Integration `*IT` tests (Docker/Keycloak, shared ports) stay serial; `-T` never parallelises within a module.

## Summary by CodeRabbit

* **Build Improvements**
  * Maven builds now use one thread per CPU core for faster execution.
  * Updated the Maven distribution used by the project.

* **Bug Fixes**
  * Improved reliability of HTTP-based test scenarios by standardizing on HTTP/1.1 and retrying transient failures.
  * Expanded validation for invalid and potentially unsafe network targets, including IPv6 addresses.

* **Tests**
  * Added broader coverage for request equality, route merging, optional values, overrides, and security-related behavior.

### CODE REVIEW

**Comment by cuioss-oliver** [comment]
@coderabbitai review

File: nifi-cuioss-rest-processors/src/test/java/de/cuioss/nifi/rest/handler/GatewayRequestHandlerTest.java [hunk 1]
```diff
@@ -132,14 +133,64 @@ private HttpRequest.Builder requestBuilder(String path) {
...
+            try {
+                return httpClient.send(request, handler);
+            } catch (IOException e) {
+                last = e;
+                httpClient = newHttpClient();
+                Thread.sleep(50L * attempt);
+            }
+        }
+        throw last;
+    }
```
[the remark this answers is not in the document]
Comment (cuioss-oliver) [reply nifi-cuioss-rest-processors/src/test/java/de/cuioss/nifi/rest/handler/GatewayRequestHandlerTest.java:166]: Good catch — applied in 83ebd473. Since the HTTP/1.1 pin is the actual fix, the JDK HttpClient discards the reset pooled connection and opens a fresh one on the next `send`, so `sendWithRetry` now retries on the existing client instead of recreating it (no more selector/manager thread accumulation). Verified stable 3× (112 tests, 0 failures per run).
